core/src/main/java/com/netflix/conductor/core/execution/tasks/SubWorkflow.java

```diff
@@ -114,7 +114,10 @@ public void start(Workflow workflow, Task task, WorkflowExecutor provider) {
 			task.getOutputData().put(SUB_WORKFLOW_ID, subWorkflowId);
 
 			// Set task status based on current sub-workflow status, as the status can change in recursion by the time we update here.
-			Workflow subWorkflow = provider.getWorkflow(subWorkflowId, false);
+			Workflow subWorkflow = provider.getWorkflow(subWorkflowId, false);			
+			if (subWorkflow.getStatus() == WorkflowStatus.COMPLETED) {
```

Comment:
@sukruthiarjula While this definitely gets the job done for the best-case scenario, there are other cases it misses. Consider how the code does this check/copy over of the payload as seen on lines 135 to 142.

Using the isTerminal function will cover more scenarios than just the COMPLETED scenario and then making sure to account for external payload storage is crucial as well.

Perhaps those lines (135-142) can be refactored out into a function of its own to be called by both the execute function and the start function. just a thought.

Reply:
Hi @pweatherby-ccctc,

As discussed, I have made the mentioned changes.